jwtauth.go

```diff
@@ -21,6 +21,36 @@ var (
 	ErrExpired      = errors.New("jwtauth: token is expired")
 )
 
+// An ExtractorFunc retreives the token string from the request.
+type ExtractorFunc func(r *http.Request) string
+
+var (
+	// ExtractCookie tries to retreive the token string from a cookie named "jwt".
+	ExtractCookie = func(r *http.Request) string {
```

Comment:
How about names like..

`FindCookieToken = func(r *http.Request) string`, `FindHeaderToken` etc..

and as you suggested, we can type the func type and in the constructor just have `findTokens ...func(*http.Request) string`

Reply:
or `findTokenFns ...func(*http.Request) string`